raiden/tests/unit/rif_comms/test_rif_comms_client.py

```diff
@@ -159,3 +153,36 @@ def test_send_lumino_message(self):
                 message=Msg(payload=str.encode(json.dumps(some_raiden_message)))
             )
         )
+
+    @pytest.mark.skip(reason="ignore")
+    def test_two_clients_topic_subscription(self):
+        notification_stream_1 = self.rif_comms_client1.connect()
+
+        notification_stream_2 = self.rif_comms_client2.connect()
+
+        one_own_sub = self.rif_comms_client1.subscribe_to(LUMINO_1_ADDRESS)
+        one_two_sub = self.rif_comms_client1.subscribe_to(LUMINO_2_ADDRESS)
+
+        two_own_sub = self.rif_comms_client2.subscribe_to(LUMINO_2_ADDRESS)
+        two_one_sub = self.rif_comms_client2.subscribe_to(LUMINO_1_ADDRESS)
+
+        peer_id_1 = self.rif_comms_client2._get_peer_id(LUMINO_2_ADDRESS)
+
+    @pytest.mark.skip(reason="ignore")
+    def test_two_clients_listen_same_topic(self):
+
+        notification_stream_1 = self.rif_comms_client1.connect()
+
+        notification_stream_2 = self.rif_comms_client2.connect()
+
+        one_own_sub = self.rif_comms_client1.subscribe_to(LUMINO_1_ADDRESS)
+        two_one_sub = self.rif_comms_client2.subscribe_to(LUMINO_1_ADDRESS)
+
+        peer_id_1 = self.rif_comms_client2._get_peer_id(LUMINO_2_ADDRESS)
+
+        for resp in one_own_sub:
+            print("Respone for 1: ", resp)
+
+        for resp in two_one_sub:
+            print("Respone for 2: ", resp)
```

Comment:
this will hang indefinitely.

with the current RIF Comms implementation, we are expecting an initial response which is the result of the subscription call (this contains no data). after it, we can expect messages to be received in this `for` loop.

what we can do here is count the amount of messages we expect to receive, and `break` out of this loop once we reach that amount.

in this case there's 0 data messages; there's just 1 message for the subscription result. if this works, i would extend this test to send 1 message to the topic, and exit each loop after 2 iterations.

Reply:
agree with the count approach for now

Reply:
resolved through #166